pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -17,11 +17,218 @@ limitations under the License.
+		t.Fatal(err)
+	}
+
+	// note that we expect a failure here
+	if err := injector.Inject(binding, creds[0]); err == nil {
+		t.Fatal("Injecting over the same binding succeeded even though it shouldn't")
+	}
+}
+
+func TestUninjectEmpty(t *testing.T) {
+	binding := createFakeBindings(1)[0]
+	injector := fakeK8sBindingInjector()
+	if err := injector.Uninject(binding); err == nil {
+		t.Fatal("Uninject empty expected error but none returned!")
+	}
+}
+
+func TestUninjectOne(t *testing.T) {
+	binding := createFakeBindings(1)[0]
+	cred := createCreds(1)[0]
+
+	injector := fakeK8sBindingInjector()
+	if err := injector.Inject(binding, cred); err != nil {
+		t.Fatal(err)
+	}
+	if err := injector.Uninject(binding); err != nil {
+		t.Fatal("Unexpected error when uninjecting")
+	}
+
+	if err := testCredentialsUninjected(injector, binding); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUninjectSame(t *testing.T) {
+	binding := createFakeBindings(1)[0]
+	cred := createCreds(1)[0]
+
+	injector := fakeK8sBindingInjector()
+	if err := injector.Inject(binding, cred); err != nil {
+		t.Fatal(err)
+	}
+	if err := injector.Uninject(binding); err != nil {
+		t.Fatal("Unexpected err when uninjecting:", err)
+	}
+	if err := injector.Uninject(binding); err == nil {
+		t.Fatal("Expected err when uninjecting twice but none found!")
+	}
+}
+
+func TestUninjectTwo(t *testing.T) {
+	bindings := createFakeBindings(2)
+	creds := createCreds(2)
+
+	injector := fakeK8sBindingInjector()
+	if err := injector.Inject(bindings[0], creds[0]); err != nil {
+		t.Fatal(err)
+	}
+	if err := injector.Inject(bindings[1], creds[1]); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := injector.Uninject(bindings[0]); err != nil {
+		t.Fatal("Unexpected err when uninjecting")
+	}
+
+	// test that bindings[0] is gone
+	if err := testCredentialsUninjected(injector, bindings[0]); err != nil {
+		t.Fatal(err)
+	}
+
+	//test that bindings[1] is still there
+	if err := testCredentialsInjected(injector, bindings[1], creds[1]); err != nil {
+		t.Error(err)
+	}
+
+	// test that bindings[1] is gone after uninject
+	if err := injector.Uninject(bindings[1]); err != nil {
+		t.Fatal("Unexpected err when uninjecting")
+	}
+
+	if err := testCredentialsUninjected(injector, bindings[1]); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func createFakeBindings(length int) []*servicecatalog.Binding {
+	ret := make([]*servicecatalog.Binding, length, length)
+	for i := range ret {
+		ret[i] = &servicecatalog.Binding{
+			ObjectMeta: kapi.ObjectMeta{
+				Name:      "name" + string(i),
+				Namespace: "namespace" + string(i),
+			},
+		}
+	}
+	return ret
+}
+
+func createCreds(length int) []*brokerapi.Credential {
+	ret := make([]*brokerapi.Credential, length, length)
+	for i := range ret {
+		ret[i] = &brokerapi.Credential{
+			Hostname: "host" + string(i),
+			Port:     "123" + string(i),
+			Username: "user" + string(i),
+			Password: "password!@#!@#!0)" + string(i),
```

Comment:
this won't correctly add the string representation of the integer (see https://play.golang.org/p/JwGzEauysz). To do so, use [`strconv.Itoa`](https://godoc.org/strconv#Itoa)